### ISSUE TITLE
Remove Quartz dependency from Astro binding

### ISSUE THREAD

**Issue by kaikreuzer** [eclipse/smarthome#3416]
We are trying to make ESH fully Java8 CP2 compatible.
One part of this effort is to replace the use of the Quartz library (which requires a full JRE) by the new ESH scheduler implementation.
The Astro binding currently still relies on Quartz, so that it should be refactored.

**Comment by kaikreuzer** [comment]
@gerrieg Are you still willing to put this on your to-do list or shall we find someone else looking into this?

**Comment by gerrieg** [comment]
Willing yes, but i am currently very busy in my real life. I can not make a promise when i can do it. If someone else can remove Quartz, it would be a big help for me.

**Comment by cweitkamp** [comment]
I can offer my help. I am looking for something to do. But I have currently no clue how ESH scheduler works. Are there any examples or documentations for me. Where can I find the code?

Another question: Compiling the Astro binding from ESH repo with `mvn clean install` results into the following error for me. What I am doing wrong?
```
[INFO] Scanning for projects...
[INFO] Computing target platform for MavenProject: org.eclipse.smarthome.binding:org.eclipse.smarthome.binding.astro:0.9.0-SNAPSHOT @ /home/christoph/workspace_4.6/smarthome/extensions/binding/org.eclipse.smarthome.binding.astro/pom.xml
[INFO] Fetching p2.index from http://download.eclipse.org/tools/orbit/downloads/drops/R20140525021250/repository/
[INFO] Fetching p2.index from http://download.eclipse.org/tools/orbit/downloads/drops/R20140525021250/repository/
[INFO] Adding repository http://download.eclipse.org/tools/orbit/downloads/drops/R20140525021250/repository
[INFO] Fetching p2.index from http://download.eclipse.org/eclipse/updates/4.4/
[INFO] Fetching p2.index from http://download.eclipse.org/eclipse/updates/4.4/
[INFO] Adding repository http://download.eclipse.org/eclipse/updates/4.4
[INFO] Fetching p2.index from http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.9.2/
[INFO] Fetching p2.index from http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.9.2/
[INFO] Adding repository http://download.eclipse.org/modeling/tmf/xtext/updates/releases/2.9.2
[INFO] Adding repository http://download.eclipse.org/modeling/emft/mwe/updates/releases/2.8.3
[INFO] Adding repository http://download.eclipse.org/modeling/emf/emf/updates/2.11/core/R201506010402
[INFO] Adding repository http://dist.springsource.org/release/GRECLIPSE/e4.4
[INFO] Fetching p2.index from http://download.eclipse.org/releases/luna/201406250900/
[INFO] Fetching p2.index from http://download.eclipse.org/releases/luna/201406250900/
[INFO] Adding repository http://download.eclipse.org/releases/luna/201406250900
[INFO] Adding repository http://www.jupnp.org/update-site/2.2.0
[INFO] Adding repository http://hstaudacher.github.io/osgi-jax-rs-connector/5.3.0
[INFO] Adding repository http://download.eclipse.org/paho
[INFO] Adding repository http://eclipse.github.io/smarthome/third-party/target/repository
[INFO] Adding repository http://www.jmdns.org/update-site
[INFO] Adding repository http://download.eclipse.org/jetty/updates/jetty-bundles-9.x/9.2.9.v20150224
[WARNING] Target location type 'Directory' is not supported
[INFO] Resolving dependencies of MavenProject: org.eclipse.smarthome.binding:org.eclipse.smarthome.binding.astro:0.9.0-SNAPSHOT @ /home/christoph/workspace_4.6/smarthome/extensions/binding/org.eclipse.smarthome.binding.astro/pom.xml
[INFO] {osgi.os=linux, osgi.ws=gtk, org.eclipse.update.install.features=true, osgi.arch=x86}
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.eclipse.smarthome.binding.astro 0.9.0.qualifier
[ERROR]   Missing requirement: org.eclipse.smarthome.binding.astro 0.9.0.qualifier requires 'package org.eclipse.smarthome.io.net.http 0.0.0' but it could not be found
[ERROR] 
[ERROR] See http://wiki.eclipse.org/Tycho/Dependency_Resolution_Troubleshooting for help.
[ERROR] Cannot resolve dependencies of MavenProject: org.eclipse.smarthome.binding:org.eclipse.smarthome.binding.astro:0.9.0-SNAPSHOT @ /home/christoph/workspace_4.6/smarthome/extensions/binding/org.eclipse.smarthome.binding.astro/pom.xml: See log for details -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException
```

**Comment by kaikreuzer** [comment]
Thanks for the offer @cweitkamp!
[Here](https://github.com/eclipse/smarthome/tree/master/bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler) is the code of the scheduler, the [unit tests](https://github.com/eclipse/smarthome/tree/master/bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/scheduler) should give you an idea of how it works (no other documentation available, sorry).
In case of any issues, best ping @kgoderis as he is the original author of that code.

> What I am doing wrong?

You need to at least one execute the Maven build from the repo root, so that all other bundles (that the Astro binding depends on) are available in your local Maven repo afterwards.

**Comment by cweitkamp** [comment]
Thanks for those hints. That is all I need.

You're right. After execute Maven build from repo root the binding compiles without any error. I had to compile the repo root with `mvn -fae clean install` because some of the tests fail to run.

**Comment by cweitkamp** [comment]
Today I pushed a first shot in my ESH [Fork](https://github.com/cweitkamp/smarthome/tree/AstroScheduler). The code compiles and I put the .jar file in my openhab2/addons folder. I configured some events in my test environment and let it run some days.

@kaikreuzer Should I already open a PR and mark it with [WIP] in the title?

**Comment by kaikreuzer** [comment]
> Should I already open a PR and mark it with [WIP] in the title?

Yes, please do so!

**Comment by cweitkamp** [comment]
PR #3501

**Comment by maggu2810** [comment]
Fixed by #3503
